### DISCUSSION TITLE
Shiny Templates issue

### DISCUSSION THREAD

**Discussion by Rebrandsoft** [shinyorg/shiny#1435] [Q&A]
### Issue Description:

When utilizing the Shiny Lib Visual Studio Templates in a fresh Maui Blazor project with the Shiny Framework selected, an issue arises where Prism fails to create a new window. This occurs despite successful compilation, but upon running the application, an exception is thrown with the message:

```
Microsoft.Maui.Controls.InvalidNavigationException: Expected Navigation Failed. No Root Window has been created.
   at Prism.Navigation.PrismWindowManager.CreateWindow(Application app, IActivationState activationState) in D:\a\Prism\Prism\src\Maui\Prism.Maui\Navigation\PrismWindowManager.cs:line 25
   at Microsoft.Maui.Controls.Application.CreateWindow(IActivationState activationState) in D:\a\_work\1\s\src\Controls\src\Core\Application\Application.cs:line 499
   at Microsoft.Maui.Controls.Application.Microsoft.Maui.IApplication.CreateWindow(IActivationState activationState) in D:\a\_work\1\s\src\Controls\src\Core\Application\Application.cs:line 428
   at Microsoft.Maui.Platform.ApplicationExtensions.CreatePlatformWindow(Activity activity, IApplication application, Bundle savedInstanceState) in D:\a\_work\1\s\src\Core\src\Platform\Android\ApplicationExtensions.cs:line 46
   at Microsoft.Maui.MauiAppCompatActivity.OnCreate(Bundle savedInstanceState) in D:\a\_work\1\s\src\Core\src\Platform\Android\MauiAppCompatActivity.cs:line 35
   at Android.App.Activity.n_OnCreate_Landroid_os_Bundle_(IntPtr jnienv, IntPtr native__this, IntPtr native_savedInstanceState) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/obj/Release/net8.0/android-34/mcw/Android.App.Activity.cs:line 3082
   at Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PPL_V(_JniMarshal_PPL_V callback, IntPtr jnienv, IntPtr klazz, IntPtr p0) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.Runtime/JNINativeWrapper.g.cs:line 125
```

### Steps to Reproduce:

1. Create a new project in Visual Studio using Shiny Templates.
![image](https://github.com/shinyorg/shiny/assets/29470448/1b90c50d-c818-4ffd-8755-79065f04feb6)

2. In the next tab "Additional Information" configure the project as in the image below.
![image](https://github.com/shinyorg/shiny/assets/29470448/070e89c4-c029-4bfb-a3d3-51983eaf0f90)

3. Build the application.
4. Run the application.

### Expected Behavior:

Prism should successfully create a new window as specified in the code without throwing an exception.

```
namespace DS_App_V2
{
    public static class MauiProgram
    {
        public static MauiApp CreateMauiApp() => MauiApp
            .CreateBuilder()
            .UseMauiApp<App>()
            .UseMauiCommunityToolkit()
            .UseShinyFramework(
                new DryIocContainerExtension(),
                prism => prism.CreateWindow("NavigationPage/MainPage"),
                new(
#if DEBUG
                    ErrorAlertType.FullError
#else
                    ErrorAlertType.NoLocalize
#endif
                )
            )
            .ConfigureFonts(fonts =>
            {
                fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
                fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
            })
            .RegisterInfrastructure()
            .RegisterAppServices()
            .RegisterViews()
            .Build();


        static MauiAppBuilder RegisterAppServices(this MauiAppBuilder builder)
        {
            // register your own services here!
            return builder;
        }


        static MauiAppBuilder RegisterInfrastructure(this MauiAppBuilder builder)
        {
            builder.Configuration.AddJsonPlatformBundle();
#if DEBUG
            builder.Logging.SetMinimumLevel(LogLevel.Trace);
            builder.Logging.AddDebug();
#endif
            var s = builder.Services;
            s.AddJob(typeof(DS_App_V2.Delegates.MyJob));
            s.AddGps<DS_App_V2.Delegates.MyGpsDelegate>();
            s.AddGeofencing<DS_App_V2.Delegates.MyGeofenceDelegate>();
            s.AddHttpTransfers<DS_App_V2.Delegates.MyHttpTransferDelegate>();
#if ANDROID
            // if you want http transfers to also show up as progress notifications, include this
            s.AddShinyService<Shiny.Net.Http.PerTransferNotificationStrategy>();
#endif
            s.AddNotifications<DS_App_V2.Delegates.MyLocalNotificationDelegate>();
            s.AddDataAnnotationValidation();
            return builder;
        }


        static MauiAppBuilder RegisterViews(this MauiAppBuilder builder)
        {
            var s = builder.Services;

            s.AddMauiBlazorWebView();

#if DEBUG
            s.AddBlazorWebViewDeveloperTools();
#endif


            return builder;
        }
    }
}
```

### Actual Behavior:

An exception is thrown upon running the application, indicating that Prism failed to create a new window.

### Environment:

.NET Version: 8
Shiny Lib Visual Studio Templates: Latest version [2.4.0]
Visual Studio: Microsoft Visual Studio Professional 2022 (64-bit) - Current Version 17.9.4

### Additional Information:

I have ensured that all dependencies and packages are up to date, including the latest version of the Shiny Lib Visual Studio Templates.

`dotnet new --install Shiny.Templates`

### Solution Explorer overview:

![image](https://github.com/shinyorg/shiny/assets/29470448/c49eb83f-6de0-4e77-aa7f-c16f34c600b3)

**Comment by aritchie** [comment]
This is something I will look into, but do you plan on mixing XAML/Viewmodels with blazor content?  Generally, Shiny Framework is 95% prism with some extras built on top of it that is really meant for a XAML app.

**Comment by Rebrandsoft** [comment]
Thank you for your response.

I intend to integrate Blazor pages with XAML native pages. This decision stems from the complexity of certain Blazor components and a monolithic Blazor page I already have, which would require significant time and resources to convert to native Maui XAML. 90% of the pages will be in XAML. My query pertains to whether Prism or Shiny framework is necessary for leveraging additional features such as background jobs and HTTP downloads, as Prism is a tool with which I have limited familiarity.

Additionally, I’ve encountered a bug while using the Shiny templates. When clicking “create project” inside the imports file, there appears to be broken code “@endif”. Furthermore, in .NET 8, the shared folder is no longer included in the solution. As a result, in imports for .NET 8, I have to manually remove the usage of the Shared folder.

**Comment by aritchie** [comment]
2.4.1 fixes the issues you've reported.  Thank you.

You don't need Shiny Framework to use the rest of Shiny.
